client/components/plans/plan-actions/index.jsx

```diff
@@ -2,26 +2,42 @@
  * External dependencies
  */
 var React = require( 'react' ),
-	classNames = require( 'classnames' );
+	classNames = require( 'classnames' ),
+	connect = require( 'react-redux' ).connect,
+	page = require( 'page' );
 
 /**
  * Internal dependencies
  */
 var abtest = require( 'lib/abtest' ).abtest,
 	analytics = require( 'analytics' ),
+	cartValues = require( 'lib/cart-values' ),
+	cartItems = cartValues.cartItems,
 	config = require( 'config' ),
 	productsValues = require( 'lib/products-values' ),
 	isFreePlan = productsValues.isFreePlan,
 	isBusiness = productsValues.isBusiness,
 	isEnterprise = productsValues.isEnterprise,
-	cartItems = require( 'lib/cart-values' ).cartItems,
-	puchasesPaths = require( 'me/purchases/paths' );
-
-module.exports = React.createClass( {
-	displayName: 'PlanActions',
+	plansPaths = require( 'my-sites/plans/paths' ),
+	puchasesPaths = require( 'me/purchases/paths' ),
+	refreshSitePlans = require( 'state/sites/plans/actions' ).refreshSitePlans,
+	upgradesActions = require( 'lib/upgrades/actions' ),
+	upgradesNotices = require( 'lib/upgrades/notices' );
 
+var PlanActions = React.createClass( {
 	propTypes: { plan: React.PropTypes.object },
 
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.isSubmitting ) {
```

Comment:
@Tug Also, why not just call this as soon as the user selects a plan, in `handleStartFreeTrial`?